src/backend/commands/vacuum.c

```diff
@@ -2629,18 +2636,22 @@ dispatchVacuum(VacuumStmt *vacstmt, VacuumStatsContext *ctx)
 {
 	CdbPgResults cdb_pgresults;
 
+	int flags = DF_CANCEL_ON_ERROR | DF_WITH_SNAPSHOT;
+
 	/* should these be marked volatile ? */
 
 	Assert(Gp_role == GP_ROLE_DISPATCH);
 	Assert(vacstmt);
 	Assert(vacstmt->options & VACOPT_VACUUM);
 	Assert(!(vacstmt->options & VACOPT_ANALYZE));
 
+	if (!vacstmt->skip_twophase)
+		flags |= DF_NEED_TWO_PHASE;
+
+
```

Comment:
Can we remove some of these extra newlines.  Otherwise, PR looks fine to me if failed vacuum regression tests (such as vacuum.sql where there are PREPARE failures) now pass with this commit.